src/main/java/net/openhft/chronicle/queue/reader/QueueEntryReader.java

```diff
@@ -0,0 +1,6 @@
+package net.openhft.chronicle.queue.reader;
+
+public interface QueueEntryReader {
+
+    ReadResult read();
```

Comment:
Returning a ReadResult means we can indicate if an entry was found, and how many outputs that entry produced then the main loop can keep track of limits and what to do when we reach the end of the queue.